test/regress/cli/CMakeLists.txt

```diff
@@ -1943,6 +1943,7 @@ set(regress_0_tests
   regress0/strings/issue11164-double-purify.smt2
   regress0/strings/issue11168.smt2
   regress0/strings/issue11197-str-ipc.smt2
+  regress0/strings/issue11703-re-uf.smt2
```

Comment:
The PR description describes 3 fixes: wrongly identifying constants regexes, wrongly allowing regexes in theory combination, and a fix in sygus. This test is an example of the second, right?

is it possible to add tests that show the first and the third?

Reply:
This test is example of the first two issues (1) it had the incorrect rewrite, (2) if we do not rewrite, the UF over RE appears in theory combination. This regression would otherwise say "unknown" without error, but I'd prefer being extra careful which was the reason for the second fix.

The fix to sygus was necessary since 2 existing sygus regressions fail unless we prevent the sygus solver from introducing new constraints, so this is already tested in our regressions.